### PR TITLE
Do 1006

### PR DESCRIPTION
**Pull request by james-vutiliti**

### CODE REVIEW

File: templates/core-statefulset.yaml [hunk 1]
```diff
@@ -116,6 +116,10 @@ spec:
         - containerPort: 3637
           name: tcp-jmx
         {{- end }}
+        # lifecycle:
```
Comment (stewiezc) [inline templates/core-statefulset.yaml:119]: is this necessary if commented out?

File: values.yaml [hunk 2]
```diff
@@ -7,10 +7,10 @@ name: "neo4j"
 
 # Specs for the Neo4j docker image
 image: "neo4j"
-imageTag: "4.2.2-enterprise"
+imageTag: "4.2.3-enterprise"
```
Comment (stewiezc) [inline values.yaml:10]: Are there any other changes required if we upgrade?

File: values.yaml [hunk 3]
```diff
@@ -7,10 +7,10 @@ name: "neo4j"
...
 imagePullPolicy: "IfNotPresent"
 # imagePullSecret: registry-secret
-acceptLicenseAgreement: "no"
+acceptLicenseAgreement: "yes"
```
Comment (stewiezc) [inline values.yaml:13]: I've been overwriting values using argocd, so some of these may not apply or are redundant. In this case it is redundant.

File: templates/core-statefulset.yaml (hunk 1 above)
Comment (james-vutiliti) [reply templates/core-statefulset.yaml:119]: Yes, it is for when we have to add .csv files into the pods from a GCP bucket.

File: values.yaml [hunk 4]
```diff
@@ -31,14 +31,18 @@ affinity: {}
...
+  prometheus.io/port: "2004"
+  prometheus.io/path: "/metrics"
 
 # Use password authentication
 authEnabled: true
 
 ## Specify password for neo4j user
 ## Defaults to a random 10-character alphanumeric string if not set and authEnabled is true
-# neo4jPassword:
+neo4jPassword: ""
```
Comment (stewiezc) [inline values.yaml:45]: again this is overwritten with argocd and so won't apply.

File: values.yaml (hunk 2 above)
Comment (james-vutiliti) [reply values.yaml:10]: Not that I know of.

File: values.yaml [hunk 5]
```diff
@@ -107,6 +113,22 @@ core:
...
+   #   image: "appropriate/curl:latest"
+   #   imagePullPolicy: "IfNotPresent"
+   #   volumeMounts:
+   #   - name: plugins
+   #     mountPath: /plugins
+   #   command:
+   #     - "/bin/sh"
+   #     - "-c"
+   #     - |
+   #       curl -L https://s3-eu-west-1.amazonaws.com/com.neo4j.graphalgorithms.dist/graph-data-science/neo4j-graph-data-science-1.4.1-standalone.jar -O
```
Comment (stewiezc) [inline values.yaml:128]: graph-data-science is a standard plugin we can install using the plugins value.

File: values.yaml [hunk 6]
```diff
@@ -323,4 +356,5 @@ global:
   bucketAccess:
     secretName: "neo4j-gcp-credentials"
     key: key.json
-  #gcpProject: define this in argocd
\ No newline at end of file
+  gcpProject: vu-james-celli
```
Comment (stewiezc) [inline values.yaml:359]: remove this one

File: values.yaml (hunk 5 above)
Comment (james-vutiliti) [reply values.yaml:128]: I want to say that I tried that and it didn't work. Then I talked with Troy and he said that plugin may only need to be installed in the Desktop.

File: values.yaml (hunk 3 above)
Comment (stewiezc) [reply values.yaml:13]: to clarify we can keep this as is, but be aware some of these are overwritten.

File: values.yaml (hunk 2 above)
Comment (stewiezc) [reply values.yaml:10]: let's keep the version as 4.2.2 and create a ticket to look at upgrading. I think we would make a new branch and then pull in the latest chart from upstream and attempt to merge.

Comment (stewiezc) [reply values.yaml:10]: 4.2.3 required for bloom

**Review by stewiezc — APPROVED** [review]

**Pull request merged**